pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -385,6 +387,14 @@ static void validatePartialUpsertStrategies(TableConfig tableConfig, Schema sche
         Preconditions.checkState(!schema.getDateTimeNames().contains(column),
             "INCREMENT merger cannot be applied to date time column: %s", column);
       }
+
+      if (entry.getValue() == UpsertConfig.Strategy.APPEND || entry.getValue() == UpsertConfig.Strategy.UNION) {
```

Comment:
Cache `entry.getValue()` to a local variable

Reply:
Gotcha, addressed the comment.